ruby/example_code/cloudtrail/create_trail.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# Demonstrates how to create an AWS CloudTrail trail
+
+# snippet-start:[cloudtrail.Ruby.createTrail]
+require "aws-sdk-cloudtrail" # v2: require 'aws-sdk'
+require "aws-sdk-s3"
+require "aws-sdk-sts"
+
+def create_trail_example(s3_client, sts_client, cloudtrail_client, trail_name, bucket_name)
+
+  resp = sts_client.get_caller_identity({})
+  account_id = resp.account
+
+  # Attach policy to an Amazon Simple Storage Service (S3) bucket.
+  s3_client.create_bucket(bucket: bucket_name)
+  begin
+    policy = {
+      "Version" => "2012-10-17",
+      "Statement" => [
+        {
+          "Sid" => "AWSCloudTrailAclCheck20150319",
+          "Effect" => "Allow",
+          "Principal" => {
+            "Service" => "cloudtrail.amazonaws.com"
+          },
+          "Action" => "s3:GetBucketAcl",
+          "Resource" => "arn:aws:s3:::#{bucket_name}"
+        },
+        {
+          "Sid" => "AWSCloudTrailWrite20150319",
+          "Effect" => "Allow",
+          "Principal" => {
+            "Service" => "cloudtrail.amazonaws.com"
+          },
+          "Action" => "s3:PutObject",
+          "Resource" => "arn:aws:s3:::#{bucket_name}/AWSLogs/#{account_id}/*",
+          "Condition" => {
+            "StringEquals" => {
+              "s3:x-amz-acl" => "bucket-owner-full-control"
+            }
+          }
+        }
+      ]
+    }.to_json
+
+    s3_client.put_bucket_policy(
+      bucket: bucket_name,
+      policy:
+    )
+    puts "Successfully added policy to bucket #{bucket_name}"
+  end
+
+  begin
+    cloudtrail_client.create_trail({
+                                      name: trail_name, # required
+                                      s3_bucket_name: bucket_name # required
+                                    })
+
+    puts "Successfully created trail: #{trail_name}."
+  # rescue StandardError => e
+  #   puts "Got error trying to create trail #{trail_name}:\n #{e}"
+  #   puts e
+  #   exit 1
```

Comment:
Did you mean to leave this? I'd either uncomment if it's valid or get rid of it.

Reply:
WoopS. Forgot to uncomment it. THanks